addons/hr_expense/models/hr_expense.py

```diff
@@ -1315,104 +1431,28 @@ def _do_create_moves(self):
         own_account_sheets = self.filtered(lambda sheet: sheet.payment_mode == 'own_account')
         company_account_sheets = self - own_account_sheets
 
-        moves = self.env['account.move'].create([sheet._prepare_bill_vals() for sheet in own_account_sheets])
-        payments = self.env['account.payment'].with_context(**skip_context).create([sheet._prepare_payment_vals() for sheet in company_account_sheets])
-        moves |= payments.move_id
-        moves.with_context(**skip_context).action_post()
+        moves = self.env['account.move'].create([sheet._prepare_bills_vals() for sheet in own_account_sheets])
+        payments = self.env['account.payment'].with_context(**skip_context).create([expense._prepare_payments_vals() for expense in company_account_sheets.expense_line_ids])
+        payment_moves = payments.move_id
+        for line in payment_moves.line_ids.filtered(lambda line: line.display_type == 'product'):
+            # We need to compute the price_total and price_subtotal of the baselines properly, else the unreconciliation fails
+            line.with_context(**skip_context).write({'price_subtotal': line.amount_currency, 'price_total': line.expense_id.total_amount})
```

Comment:
:vomiting_face: 

Can you create the line directly with `'price_subtotal': sign * line.amount_currency` as value instead of post-processing them?